src/txn/txn_timestamp.c

```diff
@@ -452,29 +453,34 @@ __txn_assert_after_reads(WT_SESSION_IMPL *session, const char *op, wt_timestamp_
 
     txn_global = &S2C(session)->txn_global;
 
-    __wt_readlock(session, &txn_global->rwlock);
-    /* Walk the array of concurrent transactions. */
     WT_ORDERED_READ(session_cnt, S2C(session)->session_cnt);
     WT_STAT_CONN_INCR(session, txn_walk_sessions);
+    WT_STAT_CONN_INCRV(session, txn_sessions_walked, session_cnt);
+
+    __wt_readlock(session, &txn_global->rwlock);
+
+    /* Walk the array of concurrent transactions. */
     for (i = 0, s = txn_global->txn_shared_list; i < session_cnt; i++, s++) {
-        WT_STAT_CONN_INCR(session, txn_sessions_walked);
         __txn_get_read_timestamp(s, &tmp_timestamp);
         if (tmp_timestamp != WT_TS_NONE && tmp_timestamp >= ts) {
-            __wt_readunlock(session, &txn_global->rwlock);
-            WT_RET_MSG(session, EINVAL,
-              "%s timestamp %s must be greater than the latest active read timestamp %s ", op,
+            __wt_err(session, EINVAL,
+              "%s timestamp %s must be after the latest active read timestamp %s ", op,
```

Comment:
Minor nit:

The phrasing of the message suggests that we are printing the "latest active read timestamp", but if I understand correctly, we are actually printing the first active read timestamp we find that is `>= ts`.